### PR TITLE
Bump Kubernetes to 1.24

### PR DESCRIPTION
**Pull request by lorenz**
### 1. Why is this pull request needed and what does it do?

This updates the Kubernetes client libraries to the newly-released 1.24 versions.

It adopts to one upstream API change where they renamed (and deprecated) an interface method.

### 3. Which documentation changes (if any) need to be made?
None.

### 4. Does this introduce a backward incompatible change or deprecation?
No.

### CODE REVIEW

File: plugin/kubernetes/object/object.go [hunk 1]
```diff
@@ -100,11 +100,11 @@ func (e *Empty) GetOwnerReferences() []v1.OwnerReference { return nil }
 // SetOwnerReferences implements the metav1.Object interface.
 func (e *Empty) SetOwnerReferences([]v1.OwnerReference) {}
 
-// GetClusterName implements the metav1.Object interface.
-func (e *Empty) GetClusterName() string { return "" }
+// GetZZZ_DeprecatedClusterName implements the metav1.Object interface.
+func (e *Empty) GetZZZ_DeprecatedClusterName() string { return "" }
```
[the remark this answers is not in the document]
Comment (lorenz) [reply plugin/kubernetes/object/object.go:104]: That's what the interface method in K8s is called, can't do anything about this here.

File: plugin/kubernetes/object/object.go [hunk 2]
```diff
@@ -100,11 +100,11 @@ func (e *Empty) GetOwnerReferences() []v1.OwnerReference { return nil }
...
 
-// SetClusterName implements the metav1.Object interface.
-func (e *Empty) SetClusterName(clusterName string) {}
+// SetZZZ_DeprecatedClusterName implements the metav1.Object interface.
+func (e *Empty) SetZZZ_DeprecatedClusterName(clusterName string) {}
```
[the remark this answers is not in the document]
Comment (lorenz) [reply plugin/kubernetes/object/object.go:107]: Same as above

**Review by chrisohaver — DISMISSED** [review]
~We have a bot that does this for us weekly (dependabot).  They should be updated next week.~

**Comment by chrisohaver** [comment]
> It adopts to one upstream API change where they renamed (and deprecated) an interface method.

Ah I see - They actually changed an interface method name to deprecate it?

**Comment by lorenz** [comment]
Yes, they did rename the interface method to deprecate it. It's a pretty weird way to deprecate it like this.

**Comment by chrisohaver** [comment]
Please [sign the commit](https://github.com/coredns/coredns/pull/5364/checks?check_run_id=6294244113) too.

**Review by chrisohaver — APPROVED** [review]
LGTM Thanks!

**Pull request merged**